WordPress/Classes/Utility/ContextManager.m

```diff
@@ -68,6 +70,28 @@ - (NSManagedObjectContext *const)newMainContextChildContext
     return [self newChildContextWithConcurrencyType:NSMainQueueConcurrencyType];
 }
 
+- (NSManagedObjectContext *const)sharedDerivedContextForKind:(Class)kind
+{
+    NSAssert([NSThread isMainThread], @"This API is not threadsafe. Should only run on the main thread");
+
+    NSString *key = [self derivedContextKeyForKind:kind];
+    NSManagedObjectContext *derivedContext = [self.derivedContextMap objectForKey:key];
+    if (derivedContext) {
+        return derivedContext;
+    }
+
+    derivedContext = [self newDerivedContext];
+    derivedContext.automaticallyMergesChangesFromParent = YES;
+    [self.derivedContextMap setObject:derivedContext forKey:key];
+
+    return derivedContext;
+}
+
+- (NSString *)derivedContextKeyForKind:(Class)kind
```

Comment:
For a future thought, we might decide it's a best practice for these to only be used by a certain type of object. Would it make sense to define a protocol, or some other strategy, to restrict what can get a specific derived context?

Reply:
Definitely! I thought about limiting this one to, perhaps, `LocalCoreDataService`, but felt a bit odd to couple the ContextManager with such.

(If) we end up adopting this API, we could, perhaps implementing a protocol, right here. Something like... `DerivedContextWorker` (OR something like that!!).

Still tinkering, since... MediaCoordinator is so complex, and there are so many moving parts, that a straightforward replace causes several deadlocks all over the place.

Thanks for stepping by sir!!

Reply:
> I thought about limiting this one to, perhaps, LocalCoreDataService, but felt a bit odd to couple the ContextManager with such

Totally agree. No need for the ContextManager to have awareness of other random types. I like the idea that the context manager could define a protocol that other classes could implement.  Or something like that.